vita3k/io/src/io.cpp

```diff
@@ -148,6 +149,44 @@ bool init_savedata_app_path(IOState &io, const fs::path &pref_path) {
     return true;
 }
 
+bool init_cache(IOState &io, const fs::path &base_path) {
+    io.cachefile = base_path / "cache" / "filepaths.cache";
+
+    if (fs::exists(io.cachefile)) {
+        fs::ifstream read_cache(io.cachefile);
+        for (std::string line; getline(read_cache, line);) {
+            std::string key = line.substr(0, line.find(','));
+            std::string value = line.substr(line.find(',') + 1);
+
+            io.cachemap.insert(std::make_pair(key, value));
+        }
+        read_cache.close();
+    } else {
+        fs::ofstream create_cache(io.cachefile);
+        create_cache.close();
+    }
+
+    return true;
+}
+
+void write_to_cache(IOState &io, const fs::path &system_path, const fs::path &icase_path) {
+    io.cachemap.insert(std::make_pair(system_path.string(), icase_path.string()));
+
+    fs::ofstream write_cache(io.cachefile, std::ios_base::app);
+    write_cache << system_path.string() + "," + icase_path.string() + "\n";
+    write_cache.close();
+}
+
+fs::path find_in_cache(IOState &io, const fs::path &system_path) {
+    const auto find_path = io.cachemap.find(system_path.string());
```

Comment:
here could use std::binary_search just in case there are lots of paths

Reply:
Requires map to be sorted XD